src/bootstrap/lib.rs

```diff
@@ -328,16 +328,22 @@ pub enum Mode {
     /// Build codegen libraries, placing output in the "stageN-codegen" directory
     Codegen,
 
-    /// Build some tools, placing output in the "stageN-tools" directory.
-    ToolStd,
-    ToolTest,
-    ToolRustc,
+    /// Build some tools, placing output in the "stageN-tools" directory. The
+    /// "other" here is for miscellaneous sets of tools that are built using the
+    /// stage0 compiler in its entirety (target libraries and all). Typically
+    /// these tools compile with stable Rust.
+    ToolOther,
+
+    /// Compile a tool which uses all libraries we compile (up to rustc).
+    /// Doesn't use the stage0 compiler libraries like "other", and includes
+    /// tools like rustdoc, cargo, rls, etc.
+    ToolAllLocalLibs,
```

Comment:
I'd prefer that we keep the split here between ToolStd/Test/Rustc since I'm hoping to use it later for https://github.com/rust-lang/rust/issues/50509 -- we want to keep `Mode` closely matched with dependency information where possible.

Looking through this though it looks like we'd only need to keep `ToolRustc` in that case (the rest would appear unused); mind renaming this to that? It would help in future tool additions too, I think, since the distinction we're making is between tools that have dependencies beyond a beta compiler; should also enable fewer rebuilds for some tools...